### PR TITLE
chore(connlib): log dynamic packet header on "unknown resource"

### PR DESCRIPTION
**Pull request by thomaseizinger**

### CODE REVIEW

File: rust/connlib/tunnel/src/dns.rs [hunk 1]
```diff
@@ -240,7 +241,10 @@ impl StubResolver {
         // as we can therefore we won't do it.
         //
         // See: https://stackoverflow.com/a/55093896
-        let question = message.first_question()?;
+        let question = message
+            .sole_question()
```
Comment (thomaseizinger) [inline rust/connlib/tunnel/src/dns.rs:245]: This is more strict than before but I think it is a good change. Happy to move to a separate PR though if requested.

**Review by conectado — APPROVED** [review]
